components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/dao/impl/AbstractDeviceDAOImpl.java

```diff
@@ -1122,4 +1128,83 @@ public EnrolmentInfo getActiveEnrolment(DeviceIdentifier deviceId, int tenantId)
         }
         return geoClusters;
     }
+
+    @Override
+    public Map<String, String> getLatestDeviceInfoMap(DeviceIdentifier deviceIdentifier, int tenantId)
+            throws DeviceDetailsMgtException {
+        Connection conn;
+        PreparedStatement stmt = null;
+        ResultSet rs = null;
+        ByteArrayInputStream bais;
+        ObjectInputStream ois;
+        Map<String, String> deviceInfoMap = null;
+        byte[] deviceInfoPayloadByteArray = null;
+        try {
+            Device device = this.getDevice(deviceIdentifier, tenantId);
+            int deviceId = device.getId();
+            String sql = "select LATEST_OPERATION_RESPONSE FROM " +
+                    "DM_DEVICE_INFO_OPERATION_RESPONSE WHERE DEVICE_ID = ?";
```

Comment:
This can be taken from the existing DM_OPERATION_RESPONSE table by getting the last completed device info operation from the enrolment-operation mapping table

Reply:
We have done modifications to the Android agent to prevent sending same information again and again. Therefore in the DM_DEVICE_OPERATION_RESPONSE table having just the modified information in previous occation.